lib/diffArr.js

```diff
@@ -0,0 +1,3 @@
+module.exports = function diffArr(base, target) {
+  return base.filter(function(item){return target.indexOf(item) < 0});
+}
```

Comment:
```suggestion
}

```

Reply:
Yes. But I haven't heard of it before, my pull request and `PropTypes.extra` have the same goal that tell developers should give outer props warning if not validate in  propsTypes. So developers can know how many props actually we use, and remove unnecessary props.

Reply:
right - but i think doing that by default would be quite noisy, and a breaking change, whereas the component author already has the ability via `exact` to achieve the same goal.

Reply:
well,  I have a question:
If you add `extra` in your project, are you both have `extra` and `no extra` components ?
I think those who notice extra, are sure to want to check all props 

Even we have `React.memo` and `PureComponent`, It will still cause re-render when unsed props change. 

And by the way, what is prop-types aim? It's only check the props you use or check the props from outer?

the `PropTypes.x.isRequired`  warn developer I must have value from parent. why we shouldn't warn developer that props not validated in child?

Reply:
(exact, not extra)

Yes, this is true - generally that's why you'd want to use a linter rule to require all your components use exact props. There are many components - like react-router's `Link` - that would break with this restriction.

Reply:
My consideration is not enough, thank you for giving me a negative example and your patience. 
Do you think should we add exact feature in prop-types?so we don't have to introduce two libraries. 
```
import PropTypes, { exact } from 'prop-types';
```

Reply:
It’s already there - the separate library just predates PropTypes.exact

Reply:
Now I think I should close this pr. My careless about the documentation.
thx.